ipatests/test_webui/test_host.py

```diff
@@ -407,11 +435,353 @@ def test_indirect_associations(self):
+        """ Test field validations for adding """
+        self.init_app()
+
+        # wrong hostname input
+        hosts_tests = [host.hostname_tilde,
+                       host.hostname_trailing_space,
+                       host.hostname_dash]
+        for hostname_test in hosts_tests:
+            self.add_record(ENTITY, hostname_test, negative=True)
+            dialog_info = self.get_dialog_info()
+            if host.BAD_HOSTNAME_MSG in dialog_info['text']:
+                self.dialog_button_click('cancel')
+                self.dialog_button_click('cancel')
+
+        # leading space in hostname
+        self.add_record(ENTITY, host.hostname_leading_space, negative=True)
+        dialog_info = self.get_dialog_info()
+        if host.BAS_HOSTNAME_SPACE_MSG in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+
+        # empty hostname
+        self.add_record(ENTITY, host.empty_hostname, negative=True)
+        hostname_i = self.find('.widget[name="hostname"]', By.CSS_SELECTOR)
+        self.assert_field_validation_required(parent=hostname_i)
+        self.dialog_button_click('cancel')
+
+        # empty domain
+        self.add_record(ENTITY, host.empty_domain, negative=True)
+        dnszone_i = self.find('.widget[name="dnszone"]', By.CSS_SELECTOR)
+        self.assert_field_validation_required(parent=dnszone_i)
+        self.dialog_button_click('cancel')
+
+        # Wrong IP input
+        ip_tests = [host.ip_alpha, host.ip_many_oct,
+                    host.ip_bad_oct, host.ip_special_char]
+        for ip_test in ip_tests:
+            self.add_record(ENTITY, ip_test, negative=True)
+            bad_ip = self.find('.widget[name="ip_address"]', By.CSS_SELECTOR)
+            self.assert_field_validation(host.BAD_IP_MSG, parent=bad_ip)
+            self.dialog_button_click('cancel')
+
+    @screenshot
+    def test_details_input(self):
+        """ Test text fields in details page """
+        self.init_app()
+
+        self.add_record(ENTITY, self.data2)
+        self.navigate_to_record(self.data2['pkey'], entity=ENTITY)
+
+        # modify
+        modify_tests = [host.mod_desc, host.mod_locality,
+                        host.mod_location, host.mod_platform,
+                        host.mod_os]
+        for mod_test in modify_tests:
+            self.fill_fields(mod_test)
+            self.button_click('save')
+            self.assert_notification()
+            self.close_notifications()
+
+        self.fill_fields(host.mod_desc_m)
+        desc_parent = self.find('.textarea-widget[name="description"]',
+                                By.CSS_SELECTOR)
+        self.button_click('undo', parent=desc_parent)
+
+        self.delete(ENTITY, [self.data2])
+
+        # otp set_otp
+        otp_tests = [host.otp_alpha, host.otp_num, host.otp_alphanum,
+                     host.otp_special, host.otp_mixed]
+        for otp_test in otp_tests:
```

Comment:
I'm not sure if whole "add", "set OTP", "delete" cycle is needed. Wouldn't just cycle in "set OTP" be enough? Resp. use reset OTP functionality for second and later and thus cover also that use case in the test so one of the subsequent test won't be needed.

Reply:
in test cases which I automate is "set" not "reset" so I would like to keep this all cycle, for reset there is another part of test case